docs(css): update random() syntax to match latest specification (#44761)
mdn/content: files/en-us/web/css/reference/values/random/index.md

random(0, 100, 10)
random(0rad, 1turn, 30deg)

/* With <random-key> */
random(auto, 0, 360)
random(property-scoped, 0s, 5s)
random(--unique-base, 400px, 100px)
random(fixed 0.5, 1em, 40vw)
random(--unique-base property-scoped, 100dpi, 300dpi)

/* With <random-key> and step interval */
random(property-scoped, 0deg, 360deg, 45deg)
random(--my-base, 1em, 3rem, 2px)
```

### Parameters

> - `<random-key>` {{optional_inline}}

Reviewer: This section should have `<DT>` should be followed by three `<DD>`s: `auto`, `fixed <number>`, and `<random-key-cache>`. The `<random-key-cache>` should explain that the value is either `element-scoped`, `<dashed-ident>` and/ or a `<*-scoped>` keyword, with a `<DT>` explaining the valid values.

  - : Controls which `random()` functions in the document will share a random base value and which will get distinct values.

>     This is one of the following values. The `*-scoped` keywords can optionally be combined with a `<dashed-ident>`, separated by a space:

Reviewer: needs work

    - `auto`

>       - : The random function generates independent random values by default. The random cache name, and thus the result, varies across every `random()` instance in a multi-component value, across different properties, and across different elements. This default `<random-key>` value is equivalent to specifying `element-scoped property-index-scoped`.

Reviewer: ```suggestion
      - : The equivalent of `element-scoped property-index-scoped`, generates independent random values across all properties and across elements. This is the default `<random-key>`.
```

    - `element-scoped`

>       - : Adds an element-specific identifier to the random cache name, so different elements get different random values.

Reviewer: Isn't the important part that the properties for the each individual element get the same seed? The spec is for the browser, but the author, i think, may be more interested in knowing the same seed is used for all the properties in the CSS block, so using `element-scoped, 50px, 300px` for both `height` and `width` will create a square. So something like:

```suggestion
      - : Adds an element-specific identifier to the random cache name, with each element getting a random value used across all element-scoped random function occurrences, with each value being independent of other elements' random values.
```

if we keep this, change different to independent

```suggestion
      - : Adds an element-specific identifier to the random cache name, so different elements get independent random values.
```
They are independent of each other, but may still generate the same value.

These suggestions apply to all `*-scoped` definitions

    - `property-scoped`

>       - : Adds the property name to the random cache name, so different properties get different random values. Shorthand declarations use the shorthand property's name.

Reviewer: move the shorthand part to the description.

    - `property-index-scoped`

>       - : Adds the property name and the index of the `random()` function among all random functions used in the same property value to the random cache name, so multiple instances in the same declaration each get different random values.

Reviewer: ```suggestion
      - : Each `random()` instance in a declaration gets an independent random value. Adds the property name and the index of the `random()` function, along with the property value, to the random cache name.
```

## Description

The `random(SEED, MIN, MAX, STEP)` function specifies the minimum and maximum values and optional step increments, starting at the minimum value. The function generates a random result within the range specified. The seed, an [optional `<random-key>`](#random-key) parameter, enables sharing or varying random base values across different properties and elements.

The minimum, maximum and step values specified must be of the same data type for the function to be valid. While the units in the two to three `<calc-sum>` parameters don't need to be the same, they do need to be of the same data type, such as {{cssxref("number")}}, {{cssxref("percentage")}}, {{cssxref("length")}}, {{cssxref("angle")}}, {{cssxref("time")}}, or {{cssxref("frequency")}}, to be valid.

### Random base value

The random base value works like a [seed for randomness](/en-US/docs/Glossary/RNG). It's a starting number that is used to generate the final random result. When two `random()` functions share the same base value, their results vary together in a predictable pattern. When they have different base values, their results are completely independent of each other.

> The optional first `<random-key>` parameter controls how the random base value is shared. Sharing enables reusing the same randomly generated value, which is needed for some design effects. The value can be explicitly set to use `auto`, a scope keyword of `element-scoped`, `property-scoped`, or `property-index-scoped`, a custom {{cssxref("dashed-ident")}} , or a `fixed <number>`, or a space-separated scope keyword and custom `<dashed-ident>` pair.

Reviewer: Changing the order makes it slightly more concise and easier to follow.

```suggestion
The optional first `<random-key>` parameter controls how the random base value is shared. Sharing lets you reuse the same randomly generated seed value, which some design effects require. The value can be explicitly set to `auto`, a custom {{cssxref("dashed-ident")}}, a `fixed <number>`, or a space-separated list of scope keywords, including `element-scoped`, `property-scoped`, or `property-index-scoped`,  optionally paired with a custom `<dashed-ident>`.
```

}
```

#### Setting both `<dashed-ident>` and `property-scoped`

> Combining a `<dashed-ident>` with `property-scoped` (e.g., `random(--custom-name property-scoped, 0, 100)`) shares the random base value across all elements for each individual property, but not across properties. Given the following, `.a`, `.b`, and `.c` will all be rectangles of the same size, because all elements share the same width value and the same height value, but width and height get independent random values:

Reviewer: ```suggestion
Combining a `<dashed-ident>` with `property-scoped` (e.g., `random(--custom-name property-scoped, 0, 100)`) shares the random base value across all elements for each individual property, but not across properties. Given the following, `.a`, `.b`, and `.c` will all be rectangles of the same size: all elements share the same `width` value and the same `height` value, but the functions within the `width` and `height` declarations return independent random values:
```
